kiwi/runtime_checker.py

```diff
@@ -239,8 +239,10 @@ def check_boot_description_exists(self):
         image_types_supporting_custom_boot_description = ['oem', 'pxe']
         build_type = self.xml_state.get_build_type_name()
         initrd_system = self.xml_state.get_initrd_system()
-        if (initrd_system == 'kiwi' and
-                build_type in image_types_supporting_custom_boot_description):
+        if (
+            initrd_system == 'kiwi' and build_type in
+            image_types_supporting_custom_boot_description
+        ):
```

Comment:
Same question, in Python `if`s are not expected to be like this : )

Reply:
I know but in the rare cases where if conditions are long the official pythonic way is to embed the expression into brackets. At least that's what people told me in the past. And yes I usually avoid the long conditions

Reply:
The long line is a fair complain, but now there are usually two options to fix the complain (not counting the ones that will create new variables for subexpressions of the condition, as you did in some cases, that is OK).

* use parens and split, and indent one char to the left to avoid confusion (optional)

```python
def xx():
    initrd_system = ''
    build_type = ''
    image_types_supporting_custom_boot_description = []
    if True:
        if (initrd_system == 'kiwi' and
           build_type in image_types_supporting_custom_boot_description):
            pass
```

* no use parens but use backslash:

```python
def xx():
    initrd_system = ''
    build_type = ''
    image_types_supporting_custom_boot_description = []
    if True:
        if initrd_system == 'kiwi' and \
           build_type in image_types_supporting_custom_boot_description:
            pass
```

Both snippets have 0 complains in the same setup with the last version.

Reply:
if you use backslash you got the complain that using backslash is redundant. Sorry what's your point here ? Do you think I start touching this if flake8 would be happy ? This type of review to prove me wrong is weird. Do the following:

```
git checkout master
tox -e check
```

see all the complains. Now checkout the branch which fixes them

```
git checkout flake_cleanup
tox -e check
```

See all fixed and used for this PR.

Please stop constructing examples out of scope it's not helpful

Thanks

Reply:
btw, both of your suggestions caused flake complains. First version complains about line break after operator (and). Second version complains about backslash being redundant between brackets. As I said I touched this for a reason

Reply:
> if you use backslash you got the complain that using backslash is redundant. Sorry what's your point here ? 

My point is to help the project.

> Do you think I start touching this if flake8 would be happy ?

Actually yes. Mostly because I tested it before commenting.

I do not know how did you mess with your system. You must to have some random `.flake8.` in your `~` or `~/.config` that you are not providing into the project.

Please, before being so rude test this is a different machine:

```
git checkout master
python3 -m venv venv
source venv/bin/activate
pip install flake8
# This is a variant of what you have in tox.ini to check only one file
flake8 --statistics -j auto --count kiwi/runtime_checker.py
```

Output:

```
kiwi/runtime_checker.py:242:17: W504 line break after binary operator
kiwi/runtime_checker.py:431:17: W504 line break after binary operator
2     W504 line break after binary operator
2
```

Change the line 242 to exactly this, to avoid the first W504:

```python
        if initrd_system == 'kiwi' and \
           build_type in image_types_supporting_custom_boot_description:
```
Output:

```
flake8 --statistics -j auto --count kiwi/runtime_checker.py
kiwi/runtime_checker.py:431:17: W504 line break after binary operator
1     W504 line break after binary operator
1
```

The second case can be fixed in the same way:

```python
        if self.xml_state.get_build_type_name() == 'iso' and \
           self.xml_state.build_type.get_flags() != 'dmsquash':
```
```
flake8 --statistics -j auto --count kiwi/runtime_checker.py
0
```

> Please stop constructing examples out of scope it's not helpful

Really. I am testing your code.

> Thanks

You are welcome.

> btw, both of your suggestions caused flake complains.

You are doing something wrong for one of the cases then, or you are providing a bad .flake8 that I do not have.

Reply:
Apologizes if I sounded rude, wasn't my intention. Changes like that in flake really brings no smile on my face. Ok I have applied your change but I'm sorry it complains:

```python
if (
    initrd_system == 'kiwi' and \
        build_type in image_types_supporting_custom_boot_description
    ):
```

```
flake8 --statistics -j auto --count kiwi/runtime_checker.py
kiwi/runtime_checker.py:243:41: E502 the backslash is redundant between brackets
1     E502 the backslash is redundant between brackets
```

I'm using tox to avoid any host side effects and the setup looks clean to me.
The files you mentioned .flake8 do not exist on my system, nowhere

I also wrote after testing, believe me :)

Frankly speaking I more trust the tox env build up from the .virtualenv.requirements.txt file because that includes all the python modules which will be present when the tool runs later on. The env as you setup it only has flake8 and maybe that's not enough. I also use the --upgrade switch and env contains:

pip list | grep flake

```
flake8                   3.6.0     
pyflakes                2.0.0  
```

So I still think the proposed changes here are ok

Thoughts ?

Reply:
So maybe you give my suggested tox command from one of the previous comments a try too ?

Reply:
I also tested the variant without brackets:

```
if initrd_system == 'kiwi' and \
    build_type in image_types_supporting_custom_boot_description:
```

which complained:

```
flake8 --statistics -j auto --count kiwi/runtime_checker.py
kiwi/runtime_checker.py:243:13: E125 continuation line with same indent as next logical line
1     E125 continuation line with same indent as next logical line
1
```

Reply:
I mean, sure, I understand the bad feeling. Please, be sure that I want to help and not complain gratuitously or bother you in any way.

For the specific question, the problem is in the parenthesis, you are providing them meanwhile I do not add them.

I put the line several times, but let me copy again:

This is OK

```python
        if initrd_system == 'kiwi' and \
           build_type in image_types_supporting_custom_boot_description:
```

This is not OK

```python
if (
    initrd_system == 'kiwi' and \
        build_type in image_types_supporting_custom_boot_description
):
```

Reply:
I know about E125 and always solved it by using brackets which imho is the recommendation by python pragma too

Reply:
I consider this a matter of taste and both is accepted by flake8

Reply:
> I also tested the variant without brackets:
> 
> ```
> if initrd_system == 'kiwi' and \
>     build_type in image_types_supporting_custom_boot_description:
> ```

As you can see, this is again not my line, as there is an extra space in the second line.

Before entering in a war of editors, please configure your editor to run flake8 as you type. If you are using emacs I can share my configuration file.

Reply:
> I consider this a matter of taste and both is accepted by flake8

Sure, but please, consider that in Python `if` do not use parenthesis. IMHO this put the valance in one place.

Reply:
ok, you have a point here. I'll adapt the code